infra/feast-operator/internal/controller/featurestore_controller_test.go

```diff
@@ -751,6 +733,89 @@ var _ = Describe("FeatureStore Controller", func() {
 			Expect(repoConfig).To(Equal(testConfig))
 		})
 
+		It("should properly set container env variables", func() {
+			By("Reconciling the created resource")
```

Comment:
can we add an env variable test for the following type of envvar?
```yaml
     env:
        - name: MY_POD_NAMESPACE
          valueFrom:
            fieldRef:
              fieldPath: metadata.namespace
```

Reply:
Done

Reply:
Shouldn't we also support `envFrom` options to read from `Secrets` or `ConfigMap`?
@tchughesiv @tmihalac

Reply:
@dmartinol that's what this added test will confirm support for... `EnvVarSource` contains fieldRef, as well as `configMapKeyRef` & `secretKeyRef`

Reply:
It is supported, there is just no specific test for that
I don't see a lot of added value in adding those tests since the only thing we are doing is overriding the original EnvVar object from the container with the ones from the spec if they have the same name

Reply:
> @dmartinol that's what this added test will confirm support for... `EnvVarSource` contains `fieldRef`, as well as `configMapKeyRef` & `secretKeyRef`

I've added a test for `fieldRef`, do you think we should add  for `configMapKeyRef` & `secretKeyRef` as well? 
see my previous comment

Reply:
@tmihalac imo, we don't need a test for the others